### PR TITLE
94 update delete ziheng

### LINKED ISSUE #94: update delete feature
#  update delete feature

# User Story

*"I want to be able to delete recipes I created, and I would like to save recipes".*

### LINKED ISSUE #95: [IMPLEMENT] Populate small recipe cards
# Feature Name

Recipe cards need to be populated with the data fetched from API. 

# Additional Context

Comments have been left in main.js specifically mentioning you. In short:

Completed: a recipe card is made using RecipeCard.js and is attached to the DOM. This recipe-card object is completely empty and needs to be populated. 

Your task: Edit RecipeCard.js (several lines or blocks of code) and main.js (less than a block of code) to populate the recipecard with any information (for example, a simple api-fetch for "cookies", or some json in local storage. You may have to create the json in local storage manually or create a tester file, and note it as a tester file).

### PR DESCRIPTION
**Pull request by hzhfred**
## Details 
Make recipeCard.js fully functional !

## New Changes
- clicking save button saves to local storage, heart icon change. 
- auto check initial state of button (whether saved or not). If saved, icon is heart at the beginning.
- delete works. (user can only see and use delete button if the recipe is user created !)
- testdeleteUpdate.js populates dummy data to allow testing.
- IMPORTANT: 
- when creating <recipe-card>, we need to add set id to the id of the recipe. Ex. <recipe-card id='spoonacular id'>. It's a good practice, also some of the functions in recipeCard.js need this id to work. 

## Linked Issue
95, 94

## Screenshots
Please include screenshots of any visual changes to the project (new HTML/CSS)

## Checklist
- [x] I have requested 2+ code reviewers
- [ ] I have reviewed and commented my own code.
- [ ] My code follows the style guidelines of this project.
- [ ] I have added new tests as necessary.
- [ ] All new and existing tests pass successfully.

### CODE REVIEW

File: source/js/components/RecipeCard.js [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
 
   set data(data) {
     // TODO: update info based on json data
     this.json = data;
-
-    const styleElem = document.createElement('style');
+    // check if it's stored in local database
+    const allSavedList = fetcherFuncs.getAllSavedRecipeId();
+    let savedList = [];
+    let createdList = [];
```
Comment (RainbowPangolin) [inline source/js/components/RecipeCard.js:18]: Make a GitHub Issue to note that this is hardcoded so we know where to look when we go back to implement the additional lists functionality.

**Review by RainbowPangolin — COMMENTED** [review]
Just create a new GitHub issue noting what will need up be updated in the future (see comment). We want to be able to quickly find what we need to modify for future changes.

File: source/index.html [hunk 2]
```diff
@@ -9,6 +9,9 @@
   <link rel="stylesheet" href="./styles/recipe-card.css">
   <link rel="stylesheet" href="./styles/createRecipe.css">
   <script defer src="https://use.fontawesome.com/releases/v5.15.4/js/all.js" integrity="sha384-rOA1PnstxnOBLzCLMcre8ybwbTmemjzdNlILg8O7z1lUkLXozs4DHonlDtnE7fpc" crossorigin="anonymous"></script>
+  <script src="https://kit.fontawesome.com/341af14cae.js" crossorigin="anonymous"></script>
```
Comment (farisashai) [inline source/index.html]: do we need this line? i thought the line above it was already importing everything from font awesome

File: source/js/testdeleteUpdate.js [hunk 3]
```diff
@@ -0,0 +1,5497 @@
...
+  // have to handle it when we open the page
+
+  // add event handler to search button to fetch recipes for query
+  // add recipes to localstorage
+  // populate search result cards
+  // navigate to search results page
+  console.log("in deleteUpdate.js");
+
+  const tempData = [
+    {
```
Comment (farisashai) [inline source/js/testdeleteUpdate.js]: this is a reallyyyy big object. maybe move to a file called "tempData.js" or something and just import it to keep things readable in this file?

**Review by farisashai — CHANGES_REQUESTED** [review]
I just merged Antonia's visual changes on the recipe card from this issue (https://github.com/cse110-fa21-group2/cse110-fa21-group2/pull/93). It looks like you're already working on the same thing here so can you add on her changes to this pull request also? It should just be updating the HTML structure to make sure everything still matches and then copying all of the code from `source/styles/recipe-card.css` into the style tag for this component.

**Comment by farisashai** [comment]
ah there's also a merge conflict to fix

File: source/js/testdeleteUpdate.js (hunk 3 above)
Comment (hzhfred) [reply source/js/testdeleteUpdate.js]: defintely

File: source/index.html (hunk 2 above)
Comment (hzhfred) [reply source/index.html]: Oh, I tried to use only the first link. it didn't work for me, my icons are not showing. I had to make an account on font awesome and include the second line to get it working.

**Pull request merged**